src/provider.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import ops
+import ops.model
+from ops.relation import ProviderBase
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: name class after the relation?
+class AlertmanagerProvider(ProviderBase):
```

Comment:
It is not clear to me who are the intended users of `AlertmanagerProvider` ? Is it integrator charms ? Or is it Prometheus ? Some docstrings here may help.

Reply:
afaiu, the provider is _always_ used _only_ by the "main" charm, i.e. Alertmanager.

Reply:
Yes Indeed. Provider is always your charm, in this case Alertmanager.